mesonbuild/interpreter/interpreter.py

```diff
@@ -3381,3 +3383,61 @@ def func_range(self, node, args: T.Tuple[int, T.Optional[int], T.Optional[int]],
         if step < 1:
             raise InterpreterException('step must be >=1')
         return P_OBJ.RangeHolder(start, stop, step, subproject=self.subproject)
+
+    @FeatureNew('response_file', '1.2.0')
+    @typed_pos_args('response_file', varargs=(str, mesonlib.File))
+    @typed_kwargs(
+        'response_file',
+        KwargInfo('prefix', str, default='@'),
+        KwargInfo('quote', (bool, str), default=True),  # TODO: choices...
+        KwargInfo('max_args_length', (bool, int), default=127),  # 127 is the limit specified in msvc documentation
+        KwargInfo('separator', str, default=' '),
+        KwargInfo('relative_paths', bool, default=True),
+    )
+    def func_response_file(self, node: mparser.BaseNode, args: T.Tuple[T.Union[mesonlib.FileOrString, T.List[mesonlib.FileOrString]]], kwargs: kwtypes.ResponseFile) -> T.Union[T.List[str], str]:
+
+        import shlex
+        from ..backend.ninjabackend import cmd_quote, gcc_rsp_quote
+
+        arguments = mesonlib.listify(list(args), flatten=True)
```

Comment:
`args[0]` should already be a flattened list.

Reply:
`args` is a tuple and `args[0]` is a list, AFAIK. That's because you could have e.g. `@typed_pos_args('foo', str, varargs=str)` where `args[0]` is a string and `args[1]` is a list of strings.